### ISSUE TITLE
Not working in Safari with `mkdocs build`

### ISSUE THREAD

**Issue by ronligt** [wilhelmer/mkdocs-localsearch#10]
When building the `site`-directory with `mkdocs build` the search-function is **not** working in Safari after loading the `html`-file. It is however working fine with `mkdocs serve`. Also tests with Firefox run fine.

## versions
mkdocs: 1.1.2
mkdocs-material: 5.5.13
mkdocs-localsearch: 0.7.0
safari: 14.0

This was tested with a bare setup of MkDocs-Material and following the instructions in the README.md.

**Comment by wilhelmer** [comment]
Thanks for reporting. I'll look into that when I'm home and on my Mac 🍎

**Comment by wilhelmer** [comment]
Okay, this is most probably related to the iframe-worker polyfill.

@squidfunk, could you help here? There seems to be an issue with [postMessage()](https://github.com/squidfunk/iframe-worker/blob/84fe3f42092b9da5cb012d9f8b6c4cba4f719455/src/worker/index.ts#L139) on Safari. 

I get the following error message (Safari 14.0 on Big Sur, latest beta):

```
[Error] Unable to post message to null. Recipient has origin null.

	postMessage
	postMessage
	(anonymous function) (search.5eca75d3.min.js:58:17668)
	dispatchEvent
	(anonymous function) (index.js:1:3105)
	promiseReactionJob
```

Also appears when using earlier versions of iframe-worker.

**Comment by squidfunk** [comment]
This is interesting. The tests run for Chrome and Firefox, but for Safari, `postMessage` seems to fail:

```
  worker
    IFrameWorker
      ✓ should initialize worker
      ✓ should terminate worker
      ✓ should delegate event handling
      ✗ should handle worker messages with listener
	Error: Timeout - Async function did not complete within 5000ms
      ✗ should handle worker messages with handler
	Error: Timeout - Async function did not complete within 5000ms
      ✓ should handle worker errors with listener
      ✓ should handle worker errors with handler
      ✗ should handle worker imports
	Error: Timeout - Async function did not complete within 5000ms
      ✓ should throw on worker options
```

All of those tests involve a call to `window.parent.postMessage`. Errors are correctly intercepted and sent to the parent window, but messages get lost. I tried to disable the following two options in the developer menu, but to no avail:

- Disable cross-origin restrictions
- Disable local file restrictions

Therefore, it more or less looks like a Webkit bug. You can open an issue over at the iframe worker repository. Unfortunately, I currently don't have the (spare) time to investigate this, but there might be a workaround, as sending errors seems to work.

**Comment by wilhelmer** [comment]
Okay, thanks for investigating. Is this a new bug in Safari 14 or did you test it with 13?

**Comment by squidfunk** [comment]
I can only test in 14 right now.

**Comment by squidfunk** [comment]
I believe I found the error (or least an error), the tests are now green:

```
Firefox 80.0 (Mac OS 10.15): Executed 9 of 9 SUCCESS (3.523 secs / 1.144 secs)
Chrome 85.0.4183.121 (Mac OS 10.15.6): Executed 9 of 9 SUCCESS (0.125 secs / 0.104 secs)
Safari 14.0 (Mac OS 10.15.6): Executed 9 of 9 SUCCESS (0.183 secs / 0.103 secs)
```

Please try [iframe-worker-0.1.7](https://github.com/squidfunk/iframe-worker/releases/tag/0.1.7) and check whether the issue persists.

**Comment by wilhelmer** [comment]
Thank you so much 🙏 Can't test right now, but maybe @ronligt can already confirm.

**Comment by ronligt** [comment]
First I would like to thank @squidfunk and @wilhelmer a lot for their help and fast respons!

I've downloaded the latest version 0.1.7 of `iframe-worker` with `npm install iframe-worker`, linked the `node_modules/iframe-worker/polyfile`-directory to `docs\js\polyfile` and changed the `theme/main.html` to:

```html
{% extends "base.html" %}
{% block config %}
{% if "localsearch" in config["plugins"] %}
<script src="{{ 'js/polyfill/index.js' | url }}"></script>
<script src="{{ 'search/search_index.js' | url }}"></script>
{% endif %}
{% endblock %}
```

I'm afraid the fix in `iframe-worker-0.1.7` did not solve the problem for `localsearch` when using `mkdocs build` in Safari. In the console I get the following message:

<img width="612" alt="Screenshot 2020-09-23 at 10 32 59" src="https://user-images.githubusercontent.com/6848987/93987732-35e80880-fd88-11ea-94ef-dd7764408f02.png">

**Comment by squidfunk** [comment]
Thanks for testing. I only had the time to fix the iframe-worker tests for now, which uncovered the issue with Safari, but I'll try to investigate this again when I find some time. I think it's best if you'd open an issue over at iframe-worker, as this has nothing to do with this project in particular.

**Comment by ronligt** [comment]
As requested by @squidfunk a new issue has been created in `iframe-worker`:

* https://github.com/squidfunk/iframe-worker/issues/100

**Comment by squidfunk** [comment]
This issue should be fixed in iframe-worker-0.1.8, see https://github.com/squidfunk/iframe-worker/issues/100#issuecomment-697534446

![Ohne Titel](https://user-images.githubusercontent.com/932156/94033671-7dd75180-fdc1-11ea-8487-ac25a7c77ba0.gif)

**Comment by ronligt** [comment]
Thank you very much @squidfunk and @wilhelmer ! When using `iframe-worker-0.1.8` the `mkdocs-localsearch` now runs in Safari using `mkdocs build` 👍
